demo_app/app/backend/directory_helper.js

```diff
@@ -8,20 +8,24 @@ export let computeDirectorySize = function(localPath) {
   let size = 0;
   let stat;
   let tempPath;
-  let contents = fs.readdirSync(localPath);
-  for (var i in contents) {
-    tempPath = localPath + '/' + contents[i];
-    stat = fs.statSync(tempPath);
-    if (stat.isDirectory()) {
-      size += computeDirectorySize(tempPath);
-    } else {
-      if (env.isFileUploadSizeRestricted && stat.size > env.maxFileUploadSize) {
-        throw new Error('File more than ' + (env.maxFileUploadSize / 1000000) + ' Mb can not be uploaded');
+  try {
+    let contents = fs.readdirSync(localPath);
```

Comment:
This operations does block the UI, in the sense, we don't allow users to do any other action in the mean time.But definitely would be better to make it async. There are few more places which uses blocking calls like this using `fs`, which can be updated in a separate PR. Will that be fine?